lib/mumukit/auth/user.rb

```diff
@@ -40,6 +40,16 @@ def apps
   end
 
   def client
+    self.class.client
+  end
+
+  def self.from_email(email)
+    user = client.users("email:#{email}").first
+    raise Mumukit::Auth::EmailNotRegistered.new('There is no user registered with that email.') unless user.present?
+    new(user['user_id'])
+  end
+
```

Comment:
Aren't you making two requests? Since you added the extra user param to the constructor, you here pass the `user` object